tests/x11/oomath.pm

```diff
@@ -20,6 +20,11 @@ use utils 'type_string_slow';
 
 sub run {
     x11_start_program('oomath');
+    if (match_has_tag('ooffice-tip-of-the-day')) {
+        # Unselect "_S_how tips on startup", select "_O_k"
+        send_key "alt-s";
+        send_key "alt-o";
```

Comment:
[DRY](https://en.wikipedia.org/wiki/Don't_repeat_yourself). You are calling the same code multiple times, please [extract method](http://refactoring.com/catalog/extractMethod.html). E.g. move that into a method of the baseclass. Two subsequent `send_key` calls without any synchronisation in between, e.g. an `assert_screen` call are most likely not really stable.

Reply:
Regarding the missing synchronization, I disagree.
The "alt-o" only requires the "alt-s" has been queued, there is no requirement it has been processed. The "alt-s" does not affect the state of the dialog in a way which requires completion before "alt-o".
In all the test runs already done, there has not been a single error with this (and I can not come up with a theoretical chain of events which would cause any errors).

Regarding the refactoring, sure.

Reply:
ok, then it's fine. So I can take a look again after the method extraction